Updated build requirements documentation (#916)
digidem/mapeo-mobile: CONTRIBUTING.md

# Contributing

Thank you so much for taking the time to contribute to Mapeo development! Mapeo-mobile is a mobile app written in react-native that currently runs on Android mobile devices. 

## Setting Up the Development Environment

To get started with Mapeo-mobile development, you need some basic software installed on your computer. Please ensure you have correct version of required software for pain-free development experience. 

### Prerequisites

* A OSX, gnu/Linux, Windows OS (64bit) computer with plenty of disk space and multi-core CPU for faster builds.  
* Node 12 (<=12.16.3)

> * NPM (Node Package Manager) or Yarn 

Reviewer: wondering if it's worth mentioning the versions of npm and yarn here. For npm, I know that Node@12 comes with npm@6 by default (latest stable is 8 – not even sure if it's compatible with Node@12). Haven't tried using yarn in this repo but my guess is that only yarn 1 will work

Author: Will need to check with @gmaclennan Last year we did transition from npm to yarn workspaces.

* Git 

We are currently using Node 12 LTS for our development work. Please refer to [React-Native documentation](https://reactnative.dev/docs/environment-setup) if you need any additional information. 

You are encouraged to use [nvm](https://github.com/nvm-sh/nvm) program to easily switch between different versions of Node.js versions.

### Build Requirements 

> For many development tasks you will also need the Android SDK installed to build Node.js for Android.

Reviewer: may be worth highlighting that some of these tools overlap with the react native setup (https://reactnative.dev/docs/0.66/environment-setup)

* Java SDK  
* Android command-line platform tools 
* Android SDK (Android API 29) 
* Android NDK (21.4.7075529)
* Andriod AVD Image (x86 ABI image for Android 10 or lower)

> * Android Studio is an optional as most tasks are done with 'gradle' build tool is available in mapeo-mobile source code itself.

Reviewer: ```suggestion
* Android Studio is optional as most tasks are done using the 'gradle' build tool, which is available through the mapeo-mobile source code itself.
```
